distributed/scheduler.py

```diff
@@ -5205,6 +5207,10 @@ def remove_plugin(self, plugin):
         """Remove external plugin from scheduler"""
         self.plugins.remove(plugin)
 
+    def register_scheduler_plugin(self, comm=None, plugin=None):
+        """TODO docstring."""
+        self.add_plugin(plugin=loads(plugin))
```

Comment:
Since this is one of the few places we run `pickle` on the scheduler, we'll want to first check the `distributed.scheduler.pickle` configuration value to make sure users haven't explicitly disallowed the scheduler from running `pickle` (which can result in arbitrary code being executed). For an example of another place where this occurs, see `Scheduler.run_function`:

https://github.com/dask/distributed/blob/b0a0eea957c8f473475a7699607643e55aecb700/distributed/scheduler.py#L6529-L6534